lib/ts/recipe/totp/types.ts

```diff
@@ -0,0 +1,232 @@
+        tenantId: string;
+        userId: string;
+        deviceName: string;
+        totp: string;
+        userContext: string;
+    }) => Promise<
+        | {
+              status: "OK";
+              wasAlreadyVerified: boolean;
+          }
+        | {
+              status: "UNKNOWN_DEVICE_ERROR" | "INVALID_TOTP_ERROR" | "LIMIT_REACHED_ERROR";
+          }
+    >;
+    verifyTOTP: (input: {
+        tenantId: string;
+        userId: string;
+        totp: string;
+        userContext: any;
+    }) => Promise<
+        | {
+              status: "OK" | "INVALID_TOTP_ERROR" | "UNKNOWN_USER_ID_ERROR";
+          }
+        | {
+              status: "LIMIT_REACHED_ERROR";
+              retryAfterMs: number;
+          }
+    >;
+};
+
+export type APIOptions = {
+    recipeImplementation: RecipeInterface;
+    config: TypeNormalisedInput;
+    recipeId: string;
+    isInServerlessEnv: boolean;
+    req: BaseRequest;
+    res: BaseResponse;
+};
+
+export type APIInterface = {
+    createDevicePOST: (input: {
+        deviceName?: string;
+        options: APIOptions;
+        session: SessionContainerInterface;
+        userContext: any;
+    }) => Promise<
+        | {
+              status: "OK" | "DEVICE_ALREADY_EXISTS_ERROR";
+              deviceName: string;
+              secret: string;
+              qrCodeString: string;
+          }
+        | {
+              status: "DEVICE_ALREADY_EXISTS_ERROR";
+          }
+        | GeneralErrorResponse
+    >;
+
+    listDevicesGET: (input: {
+        options: APIOptions;
+        session: SessionContainerInterface;
+        userContext: any;
+    }) => Promise<
+        | {
+              status: "OK";
+              devices: {
+                  name: string;
+                  period: number;
+                  skew: number;
+                  verified: boolean;
+              }[];
+          }
+        | GeneralErrorResponse
+    >;
+
+    removeDevicePOST: (input: {
+        deviceName: string;
+        options: APIOptions;
+        session: SessionContainerInterface;
+        userContext: any;
+    }) => Promise<
+        | {
+              status: "OK";
+              didDeviceExist: boolean;
+          }
+        | GeneralErrorResponse
+    >;
+
+    verifyDevicePOST: (input: {
+        deviceName: string;
+        totp: string;
+        options: APIOptions;
+        session: SessionContainerInterface;
+        userContext: any;
+    }) => Promise<
+        | {
+              status: "OK";
+              wasAlreadyVerified: boolean;
+          }
+        | {
+              status: "UNKNOWN_DEVICE_ERROR" | "INVALID_TOTP_ERROR" | "LIMIT_REACHED_ERROR";
+          }
+        | GeneralErrorResponse
+    >;
+
+    verifyTOTPPOST: (input: {
+        totp: string;
+        options: APIOptions;
+        session: SessionContainerInterface;
+        userContext: any;
+    }) => Promise<
+        | {
+              status: "OK" | "INVALID_TOTP_ERROR" | "UNKNOWN_USER_ID_ERROR";
```

Comment:
This should also have the max and current attempt counts (until `LIMIT_REACHED_ERROR`) for status `INVALID_TOTP_ERROR`

Reply:
Added them